provider/testdata/src/lib.rs

```diff
@@ -128,73 +156,76 @@ pub mod versions {
 #[cfg(feature = "std")]
 pub mod paths;
 
-#[cfg(feature = "static")]
-use {
-    icu_provider_adapters::fallback::LocaleFallbackProvider, icu_provider_blob::StaticDataProvider,
-};
+use icu_provider::prelude::*;
+use icu_provider_adapters::fallback::LocaleFallbackProvider;
 
-#[cfg(feature = "fs")]
-use icu_provider_fs::FsDataProvider;
-
-/// Get a data, loading from the test data JSON directory.
-///
-/// You can optionally specify your own test data with the
-/// `ICU4X_TESTDATA_DIR` environment variable.
+/// A data provider that is compatible with all ICU `_unstable` constructors.
 ///
-/// # Panics
+/// The return type of this method is not considered stable, mirroring the unstable trait
+/// bounds of the constructors. For matching versions of `icu` and `icu_testdata`, however,
+/// these are guaranteed to match.
+pub fn unstable() -> LocaleFallbackProvider<UnstableDataProvider> {
+    // The statically compiled data file is valid.
+    #[allow(clippy::unwrap_used)]
+    LocaleFallbackProvider::try_new_unstable(unstable_no_fallback()).unwrap()
+}
+
+/// A data provider that is compatible with all ICU `_unstable` constructors.
 ///
-/// Panics if unable to load the data.
-// The function is documented to allow panics.
-#[allow(clippy::panic)]
-#[cfg(feature = "fs")]
-pub fn get_json_provider() -> FsDataProvider {
+/// The return type of this method is not considered stable, mirroring the unstable trait
+/// bounds of the constructors. For matching versions of `icu` and `icu_testdata`, however,
+/// these are guaranteed to match.
+pub fn unstable_no_fallback() -> UnstableDataProvider {
+    UnstableDataProvider
+}
+
+/// An [`AnyProvider`] backed by baked data.
+pub fn any() -> impl AnyProvider {
+    // The baked data is valid.
+    #[allow(clippy::unwrap_used)]
+    LocaleFallbackProvider::try_new_with_any_provider(any_no_fallback()).unwrap()
+}
+
+/// An [`AnyProvider`] backed by baked data.
+pub fn any_no_fallback() -> impl AnyProvider {
+    UnstableDataProvider
+}
+
+/// A [`BufferProvider`] backed by a Postcard blob.
+#[cfg(feature = "buffer")]
+pub fn buffer() -> impl BufferProvider {
```

Comment:
Discussion: I would still prefer `icu_testdata::postcard()` over `icu_testdata::buffer()`. Even `icu_testdata::blob()` would be more precise.

Postcard is a special case of Blob\* which is a special case of Buffer.

We have two different types of buffer providers we can expose: the FsDataProvider JSON one and the BlobDataProvider Postcard one.

I also think naming the functions with their actual data source has educational benefit.

I feel less strongly about `icu_testdata::unstable()` vs `icu_testdata::baked()` but I would slightly prefer `::baked()` for the same reasons. It's also 3 characters shorter.

\* I realize that Blob supports only Postcard at this time.

Reply:
The idea here is to make them symmetric with the constructors. First time users will not know that `::baked()` implements the traits required by unstable, or that `::blob()` is a `BufferProvider`. We're exposing `fn buffer() -> impl BufferProvider`, and I don't think anyone needs to care what the buffer format is.

Reply:
I get the idea; I just don't agree with the weighting of priorities.

But I won't block this PR on it. I'd like to discuss later but it'll be an easy refactor-rename to change it.